File: parser/src/main/java/io/ticofab/androidgpxparser/parser/domain/Email.java
package io.ticofab.androidgpxparser.parser.domain;

public class Email {

    private String mId;
    private String mDomain;

    private Email(Builder builder) {
        this.mId = builder.mId;
        this.mDomain = builder.mDomain;
    }

    public String getId() {
        return mId;
    }

    public String getDomain() {
        return mDomain;
    }

    public static class Builder {
        private String mId;
        private String mDomain;

        public Builder setId(String id) {
            mId = id;
            return this;
        }

        public Builder setDomain(String domain) {
            mDomain = domain;
            return this;
        }

        public Email build() {
            return new Email(this);
        }
    }
}

File: parser/src/main/kotlin/io/ticofab/androidgpxparser/parser/domain/Email.kt
package io.ticofab.androidgpxparser.parser.domain

class Email private constructor(builder: Builder) {
    val id: String?
    val domain: String?

    class Builder {
        var id: String? = null
            private set
        var domain: String? = null
            private set

        fun setId(id: String?): Builder {
            this.id = id
            return this
        }

        fun setDomain(domain: String?): Builder {
            this.domain = domain
            return this
        }

        fun build(): Email {
            return Email(this)
        }
    }

    init {
        id = builder.id
        domain = builder.domain
    }
}